---
title: "BIGX_Analysis_2020" 
author: Nikita Telkar 
date: "March 3, 2020"
output: 
  html_document: 
    keep_md: yes 
    toc: true 
    toc_float: 
      collapsed: true 
      smooth_scroll: true 
    theme: flatly  #cerulean, journal, flatly, darkly, readable, spacelab, united, cosmo, lumen, paper, sandstone, simplex, and yeti
    highlight: zenburn  
#tango, pygments, kate, monochrome, espresso, zenburn, haddock, textmate. 
---  

***  

This is the data that *Victor M* generated for the predicted and novel miRNA. I've analyzed miRNA expression by  
+ Neural Tube Defects' (NTDs) status  
+ Sex  
+ Trimester  
+ Ancestry (TBD)  

*Figures and plots to be made*  

***

## Data Loading and Visualisation:  

```{r, message=FALSE, echo =TRUE, r, error=FALSE, message=FALSE, include=FALSE} 
library(rmarkdown)
# library(installr)
library(dplyr) 
library(tidyr) 
library(readxl) 
library(DT) 
library(knitr) 
library(kableExtra) 
library(ggplot2)
library(tidyverse)
library(devtools)
library(magrittr)
library(arsenal)

raw <- read_excel("Z:/Nikita/Projects/snc_placenta/data/raw/quantification_novel_mirnas_rpm_norm.xlsx", col_names =FALSE)

pred <- read_excel("Z:/Nikita/Projects/snc_placenta/data/raw/predictions.xlsx", col_names = TRUE)

pData <- read_excel("Z:/Nikita/Projects/mirna_fetal_tissues/data/processed/miRNA_fetal_tissues_pDat.xlsx", col_names = TRUE)

print(pred, n = 15)

rob <- raw [1:32] #dropping all TCGA samples and keeping only Roblab samples
colnames(rob) = rob[1, ]
rob <- rob [-1,]


#fullJoinDf <- full_join(tableA,tableB,by=”Customer.ID”)
#band_members %>% full_join(band_instruments2, by = c("name" = "artist"))")

same <- pred %>% 
              inner_join(rob, by = c ("precursor_name" = "precursor")) #joining and keeping only common columns from pred and rob

#Doesn't work because rob has teo predicted miRNAs with the same precursor name

#same <- na.omit(same)

#same <- same %>%
 #        drop_na()
  
#same <- unique(same)

#same <- same %>% 
         # distinct(precursor_name, .keep_all = TRUE) #makes no sense as two of each with the same precursor name

pred_nov <- same %>% 
            filter(Category =="Novel") %>% 
        #   distinct(precursor_name, .keep_all = TRUE) %>% 
        #   arrange(desc('novomirank_score')) %>% 
        #   filter(novomirank_score > 1.5 )
            mutate_at(vars(novomirank_score), round, digits = 1)

nov_rank <-pred_nov %>%
  group_by(novomirank_score) %>% 
  summarise(number = sum(novomirank_score))

#maybe make a new object and then plot? 
nov_rank %>%   
  ggplot(aes(x=factor(novomirank_score), y=number, fill=number)) + #using x=factor()to display all x-axis values
  geom_bar(stat = "identity") +
  scale_fill_continuous(low="coral", high="purple") +
  labs(title ="Number of Novel miRNAs by NovoMiRank Score", subtitle ="The NovoMiRank score has threshold values of 0 to 3",x = "NovoMiRank Score (z-score)", y = "Number of miRNAS") 


pred_score <-pred_nov %>%
  mutate_at(vars(probability), round, digits = 2) %>% 
  mutate(prob_percent = probability * 100)

pred_score <- pred_score %>%
  group_by(probability) %>% 
  summarise(number = sum(probability))

pred_score <-pred_score %>%
  mutate_at(vars(number), round)

pred_score <- pred_score %>% 
  mutate_at(vars(probability), funs(probability * 100)) %>% 
  mutate_at(vars(probability), funs(paste(probability, "%", sep ="")))


pred_score %>%   
  ggplot(aes(x=probability, y=number, fill=number)) +
  geom_bar(stat = "identity", width = 0.7) +
  scale_fill_continuous(low="coral", high="purple") +
#  scale_x_continuous(breaks = c(50, 60, 70, 80, 90, 100))
#  scale_y_discrete(limits=c(50, 100))
  coord_flip() +
  labs(title ="Probability of Novel miRNAs having a true precursor", x = "Probability depicted as Percentages", y = "Number of miRNAS") 

pDat <- pData %>% 
  mutate_at(vars(gscID), funs(str_replace_all(gscID, "_" , "-")))

t_rob <- t(rob)
t_rob <- as.data.frame(t_rob)
t_rob <- t_rob %>% 
  rownames_to_column("VALUE") #convert rownames to first column

match <- t_rob %>% 
  inner_join(pDat, by = c("VALUE" = "gscID"))

pDat <- match[c(1,28901:28927)] #updated pDat for the Rob lab samples only
colnames (pDat) [1] <- "sample"

match <- match[-c(28900:28927)]

colnames(match) = t_rob[2, ] #doesn't work for some ungodly reason. Unlisting it, as.character doesn't work either

mi <- t_rob[2, ] #So converting to separate object as rows

mi <- t(mi) #converting the row to a column

is.character(mi)

colnames(match) = mi[, 1] #adding the column as column names

eDat <- match #Roblab samples miRNA expression data

eDat <- eDat %>% 
  mutate_at(vars(-miRNA), funs(as.character)) %>% 
  mutate_at(vars(-miRNA), funs(as.numeric))

t_edat <- data.frame(t(eDat))
colnames(t_edat) <- t_edat[1,] #again the idiotic reason as above
cn <- t_edat[1,]
cn <-t(cn)
colnames(t_edat) <- cn[,1]
t_edat <- t_edat [-1, ]

colnames (eDat) [1] <- "sample"

rm(match)

#str(eDat)


meta_dat <- pDat %>% 
  inner_join(eDat, by = "sample")

meta_dat %>% 
  count(sex)

cn <- colnames(meta_dat)

t_meta <- rbind(cn, meta_dat)
t_meta <- t(t_meta)
colnames(t_meta) = t_meta[1, ]

#str(eDat)
      
meta_dat <- meta_dat %>%
  mutate(sex2 = recode(sex,"MALE" = 1, "FEMALE" = 2)) %>% #recoding sex to numeric values
  select(c(1:10), "sex2", everything())
  
meta_dat %>% 
  count(sex2)

```

### Phenotype Info:  

```{r, warning=FALSE, results= "asis"}
sum_stats <- pDat[c(6,9,10,28)]
tab1 <- tableby(trimester~., data = sum_stats, test =FALSE) #package: arsenal --> cannot use with %>% 
summary (tab1, title ="Breakdown of Samples by Trimester")
  
```  

### Genotype Info:  

```{r, warning=FALSE, error=FALSE, message=FALSE}
library(devtools)
install_github("vqv/ggbiplot")

library(ggbiplot)
library(limma)

#str(t_edat)

t_edat <-t_edat %>% 
  mutate_all(funs(as.character)) %>% 
  mutate_all(funs(as.numeric))
rn2 <- colnames(eDat)
rn2 <- rn2[c(-1)]
row.names(t_edat) <- rn2

paged_table(t_edat)


(pDat$sample == names(t_edat)) %>% all() #CHECKING TO MAKE SURE THAT SAMPLE IDS ARE IN ORDER

desmat_cond <- model.matrix(~condition, pDat) #linear modelling on condition --> controls vs NTD

con_mat <- lmFit(t_edat, desmat_cond)

con_mat_eb <- eBayes(con_mat)

diff_condition_genes <- topTable(con_mat_eb, n = Inf)



#Controlling for Sex

desmat_cond_sex <- model.matrix(~condition + sex, pDat) #conditionmodel controlling for sex

con_sex_mat <- lmFit(t_edat, desmat_cond_sex)

con_sex_mat_eb <- eBayes(con_sex_mat)

con_sex_mat_genes <- topTable(con_sex_mat_eb, n = Inf)


#Controlling for GA

desmat_cond_ga <- model.matrix(~condition + GA, pDat) #condition model controlling for GA

con_ga_mat <- lmFit(t_edat, desmat_cond_ga)

con_ga_mat_eb <- eBayes(con_ga_mat)

con_ga_mat_genes <- topTable(con_ga_mat_eb, n = Inf, adjust.method = "fdr", sort.by="B", resort.by = "logFC")

```

### Controls vs NTDs in Trimester 2 samples:  
(Univariate Analysis)  

Considering only Trimester 2 Samples:  

```{r}
tri2p <- pDat %>% #trimester 2 phenotype
  filter (trimester == "2")

tri2 <- as.character(tri2p$sample) #select sample names of trimester 2 --> make value

#tri2g <- t_edat %>%   #this dropped the trimester 2 samples and kept the trimester 1 and 3 --> we want the opposite
 # modify_at(tri2,~NULL)

tri2g <- t_edat %>% 
  select(which(names(t_edat) %in% tri2))

(tri2p$sample == names(tri2g)) %>% all()

tri2_mat <- model.matrix(~condition + sex, tri2p) #desmat for trimester 2 samples modelling for condition controllinf for sex

tri2_mat_sex <- lmFit(tri2g, tri2_mat)

tri2_mat_eb <- eBayes(tri2_mat_sex)

diff_tri_sex <- topTable(tri2_mat_eb, n = Inf, adjust.method = "fdr", p.value = 0.05) 
paged_table(diff_tri_sex) #control is the reference --> logFC value means it is downregulated in NTDs
```  

Now here, we need to specify that we want to consider **condition** as our variable or `coefficient` of interest. Otherwise, the function relies on the default intercept which would give us different results:  

```{r}
diff_tri_sex <- topTable(tri2_mat_eb, n = Inf, adjust.method = "fdr", coef = 2, p.value = 0.05) #specifying variable of interest as control/NTD from tri2_mat column 2
paged_table(diff_tri_sex)
```

***

### Sex-Differential Genes for the 2nd and 3rd trimester:  
(Multivariate Analysis):  

```{r}

tri23p <- pDat %>% 
  filter (!trimester == "1")

tri23 <- as.character(tri23p$sample) 

tri23g <- t_edat %>% 
  select(which(names(t_edat) %in% tri23))

(tri23p$sample == names(tri23g)) %>% all()

tri23_mat <- model.matrix(~sex + trimester, tri23p) #desmat for trimester 2 and 3 samples modelling for sex controlling by trimester --> GA doesn't work since matrix larger than no of samples, and no residual degress of freedom

tri23genes_sex <- lmFit(tri23g, tri23_mat) %>% 
  eBayes()

tri23_list <- topTable(tri23genes_sex, n = Inf, adjust.method = "fdr", coef = 2, p.value = 0.05) #differentially expressed by sex in trimester 2 and 3 in controls and NTDs
paged_table(tri23_list)

tri23_list_tri3 <- topTable(tri23genes_sex, n = Inf, adjust.method = "fdr", coef = 3, p.value = 0.05) #differentially expressed in trimester 3 in controls and NTDs
nrow(tri23_list_tri3)
paged_table(tri23_list_tri3)
```  

Making **contrasts** manually as opposed to implicit determinimation using `lmFit`:  

```{r, include=FALSE, eval=FALSE}
tri23_mat_contrast <- model.matrix(~tri23p$sex -1)
#matrix
colnames(tri23_mat_contrast) <- c("Female", "Male")

tri23_sex_contrast <- lmFit(tri23g, tri23_mat_contrast) #fit the matrix to the model

tri23_contrats_mat <- makeContrasts(sexx_diff = Female - Male, levels = tri23_mat_contrast) #make the contrats for the model

tri23_contrast <- contrasts.fit(tri23_sex_contrast, tri23_contrats_mat) %>% 
  eBayes() #fit the contrast matrix to the earlier lmfit

tri23_contrast_genes <- topTable(tri23_contrast, n = Inf, adjust.method = "fdr", p.value = 0.05) 
paged_table(tri23_contrast_genes)
#summary(decideTests(tri23_contrast_genes))
```  

However, since there is only variable we're considering above (**sex**), we don't really need to use a contrast matrix. (Specifying `coef = 2`  in `tri23genes_sex` isn't going to make a difference since there is only one variable and we aren't controlling for confoudnders or covariates)   

```{r}
tri23_list <- topTable(tri23genes_sex, n = Inf, adjust.method = "fdr", p.value = 0.05)
paged_table(tri23_list)
```  

Now, if we actually applied the same `coef` fucntion to consider *sex* as the variable of interest in the **Controls vs NTDs in Trimester 2 samples** analysis above, we get the same significantly differentially expressed miRNA by sex as when looking at **Sex-Differential Genes in Trimester 2 and 3**:  

```{r}
diff_tri_sex <- topTable(tri2_mat_eb, n = Inf, adjust.method = "fdr", coef = 3, p.value = 0.05)
paged_table(diff_tri_sex)
```  

***  


### Differential miRNA by Trimester in Controls:  

```{r}

controls_p <- pDat %>% 
  filter (condition == "con")

control_samp <- as.character(controls_p$sample) 

control_g <- t_edat %>% 
  select(which(names(t_edat) %in% control_samp))

(controls_p$sample == names(control_g)) %>% all()

controls_mat <- model.matrix(~trimester + sex, controls_p) #desmat for all controls modelling for trimester controlling by sex 

controls_genes <- lmFit(control_g, controls_mat) %>% 
  eBayes()

controls_tri2_list <- topTable(controls_genes, n = Inf, adjust.method = "fdr", coef = 2, p.value = 0.05) #genes differentially expressed in controls at trimester 2
nrow(controls_tri2_list)
paged_table(controls_tri2_list)

controls_tri3_list <- topTable(controls_genes, n = Inf, adjust.method = "fdr", coef = 3, p.value = 0.05) #genes differentially expressed in controls at trimester 3
nrow(controls_tri3_list)
paged_table(controls_tri3_list)

controls_sex_list <- topTable(controls_genes, n = Inf, adjust.method = "fdr", coef = 4, p.value = 0.05) #genes differentially expressed in controls by sex --> none?
controls_sex_list
```
