---
title: "GSC-1790 Data Type Conversion" 
author: Nikita Telkar 
date: "May 19, 2020"
output: 
  html_document: 
    keep_md: yes 
    toc: true 
    toc_depth: 4
    toc_float: 
      collapsed: false 
      smooth_scroll: false 
    theme: cosmo  #cosmo, paper, lumen, sandstone, simplex, yeti; cerulean, journal, flatly, darkly, readable, spacelab, united
    highlight: zenburn #tango, pygments, kate, monochrome, espresso, zenburn, haddock, textmate.
--- 


 ***

###  Introduction 

Here, I'll generate BAM and count matrix files from the raw FASTQ files for the GSC-1790 test samples run. I'll also generate some plots to visualise the data 

 ***

###  1.0 Loading Packages and Path/Package Corrections   

```{r, libraries, error=FALSE, warning=FALSE, message=FALSE}

library(tidyverse)
library(dplyr)
library(plyr)
library(here) 
library(rmarkdown)
library(knitr)
library(stringr)
library(DT)
library(kableExtra)
library(formatR)
library(Rsamtools)
library(Rsubread)
library(pheatmap)
library(RColorBrewer)
library(biomaRt)
library(extrafont)
loadfonts(device = "win")


if( Sys.info()['sysname'] == "Linux" ) {
  fas <- library(fastqcr)
} else {
    fas <- "library(fastqcr)"
} #load package fastqcr only if operating system is Unix

if( Sys.info()['sysname'] == "Linux" ) {
  path <- "/home/BCRICWH.LAN/nikita.telkar/RobinsonLab/Nikita/Projects/cell_sorted/Images/Phred_Scores.png"
} else {
    path <- "Z:\\Nikita\\Projects\\cell_sorted\\Images\\Phred_Scores.png"
} #filepath dependancy on operating system

``` 

```{r}
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=50),tidy=TRUE)

colpal1 <- c("#148eff", "#ff61ad", "#ffae00")
colpal2 <- c("#801361", "#e44c4a", "#ffb708")
colour_pal <- brewer.pal(11, "PiYG")

```  

```{r, include=FALSE, echo=FALSE}

#include = FALSE prevents code and results from appearing in the finished file. R Markdown still runs the code in the chunk, and the results can be used by other chunks.

#echo = FALSE prevents code, but not the results from appearing in the finished file. This is a useful way to embed figures.

#message = FALSE prevents messages that are generated by code from appearing in the finished file.

#warning = FALSE prevents warnings that are generated by code from appearing in the finished.

#fig.cap = "..." adds a caption to graphical results. 

```

 ***

###  2.0 Loading Data  

I'll be using the `Rsubread` package, since it is the only one that is compatible to be working in R.


```{r, loading BAM files}

bam_files <- (here::here('data', 'external', "Cell_Sorted_Test_Samples_08042020_GSC1790", "All_BAM_files"))

bam.files <- list.files(path = "Z:/Nikita/Projects/cell_sorted/data/external/Cell_Sorted_Test_Samples_08042020_GSC1790/All_BAM_files", full.names = TRUE)

fastq_files <- (here::here('data', 'external', "Cell_Sorted_Test_Samples_08042020_GSC1790", "All_fastq_files"))

#buildindex(basename="hg38_ref", reference=here("GRCh38_latest_genomic.fna.gz"))

```  

 ***  

###  2.0 Generating Count Matrix   

I'll be using the `featureCounts` function fro the `Rsubread` package.   

```{r, load fc saved object, include=FALSE}

fc <- readRDS(file = here::here('data', 'processed', "featurecounts_object_GSC_1790.rds"))

```

```{r, running featurecounts, eval = FALSE}

#props <- propmapped(files=bam_files)
fc <- featureCounts(bam.files, annot.inbuilt="hg38") #GSC control samples took longer to run/process

```

```{r, count matrix}

names(fc)

count_stats <- fc$stat
count_stats$Status

eDat <- fc$counts #row names are Entrez gene IDs

eDat_extra <- fc$annotation
names(eDat_extra)
head(eDat_extra, n = 2)

```  

 ***  

###  3.0 Data Preprocessing    

```{r, data cleaup edat}

genes <- row.names(eDat)

#shortening sample names
samples <- as.data.frame(colnames(eDat))
samples$`colnames(eDat)` <- str_sub(samples$`colnames(eDat)`, 1, 6)
samples <- samples$`colnames(eDat)`

colnames(eDat) <- samples
row.names(eDat) <- genes


eDat_mat <- eDat #keeping in matrix form

eDat <- as.data.frame(eDat)
dim(eDat) #28395 24

eDat2 <- eDat %>%  #removing sequences with less than 1 read count across all samples
  rownames_to_column(var = "entrezgene_id") %>% 
  mutate(Sum = rowSums(.[2:25])) %>%  
  filter(Sum >= 1) %>% 
  column_to_rownames(var = "entrezgene_id")

dim(eDat2) #22938  25

eDat2_mat <- as.matrix(eDat2)

eDat_extra$GeneID <- as.character(eDat_extra$GeneID)

head(eDat, n=3)
```  

When sequences that have zero counts (i.e. no recorded expression) across all 24 samples were removed, the number of sequences reduced from 28,395 to 22,938 (5457 sequences had counts of less than 1).  

```{r, data cleaup pdat}

pDat<- as.data.frame(samples)
colnames(pDat) [1] <- "GSC_ID"
pDat$titration <- c("500ng", "250ng", "100ng", "50ng", "25ng", "5ng", "1ng", "0ng")
pDat <- pDat %>% 
  dplyr::arrange(titration)
pDat$sample <- c("PL-290v", "TCR-2-18_v_SNAP", "FG032A")
pDat$trimester <- c("first", "term", "term")
pDat<- pDat %>% 
  dplyr::arrange(GSC_ID) %>% 
  column_to_rownames(var = "GSC_ID")

head(pDat, n=10)
```  

####  3.1 Using BiomaRt to get IDs    

The `featureCounts` function gives Entrez IDs, and so using the `biomaRt` package, I'll add the gene symbols.  

```{r, exploring biomart, eval=FALSE, include=FALSE}

filters <- listFilters(ensembl)
atributes <- listAttributes(ensembl)

searchAttributes(mart = ensembl, pattern = "hgnc")
searchAttributes(mart = ensembl, pattern = "ncRNA")
searchAttributes(mart = ensembl, pattern = "mir")
searchAttributes(mart = ensembl, pattern = "entrez")

searchFilters(mart = ensembl, pattern = "hgnc")
searchFilters(mart = ensembl, pattern = "ensembl")
searchFilters(mart = ensembl, pattern = "ncRNA")
searchFilters(mart = ensembl, pattern = "mir")

```


```{r, adding HGNC symbols, cache=TRUE, message=FALSE}

ensembl = useMart("ensembl",dataset="hsapiens_gene_ensembl")

#getting all id names
identifiers <- getBM(attributes=c("entrezgene_id", "hgnc_symbol", "ensembl_gene_id"), 
                     values = genes, mart = ensembl) 


#some gene symbols don't have gene ids?
entrez <- getBM(attributes=c("entrezgene_id", "hgnc_symbol"), 
                     values = genes, mart = ensembl)
entrez$entrezgene_id <- as.character(entrez$entrezgene_id)
entrez <- entrez %>% #removing zero expression
  na_if("") %>%
  na.omit

#getting ncRNA info
nc_ids <- getBM(attributes=c("ensembl_gene_id", "refseq_ncrna", "refseq_ncrna_predicted"), 
                values = genes, mart = ensembl)


#emsembl ids are the most exhaustive list
all_ids <- identifiers %>% 
  inner_join(nc_ids, by = "ensembl_gene_id")

str(all_ids)

all_ids$entrezgene_id <- as.character(all_ids$entrezgene_id)
```  

####  3.2 Making mDat   

```{r, making mDat, message=FALSE}

mDat <- eDat %>% 
  rownames_to_column(var = "entrezgene_id") %>% 
  inner_join(entrez, by = "entrezgene_id") %>% 
  dplyr::select("entrezgene_id", "hgnc_symbol", everything())

mDat <- mDat %>% 
  inner_join(eDat_extra, by = c("entrezgene_id" = "GeneID"))

mDat[1:5,] %>% 
  kable() %>% 
  kable_styling(bootstrap_options = c("striped", "hover", "condensed"), full_width = F, fixed_thead = T)
```   

####  3.2 Exploratory Analysis  {.tabset}  

##### 3.2.1 Histograms

```{r, histograms}

log_dat <-  log2(eDat_mat + 1)
hist(log_dat)

log_eDat2 <- log2(eDat2_mat + 1)
hist(log_eDat2)

```

##### 3.2.2 Sample Correlation
 
```{r, ea sample cor}

sample_cor <- eDat %>% 
  cor()

my_colour = list(titration = c("0ng" = "#87CEFA", "1ng" = "#87CEEB", "5ng" = "#00BFFF", 
                               "25ng" = "#1E90FF", "50ng" = "#4169E1", "100ng" = "#0000FF", 
                               "250ng" = "#00008B", "500ng" = "#0e0e3f"))
  
sample_cor %>% 
  pheatmap(cluster_rows = TRUE, clustering_distance_cols = "euclidean", 
                clustering_method = "ward.D2", show_colnames = TRUE, show_rownames = T, 
                color = colour_pal, main = "Heatmap - Ward.D2",
                annotation = pDat[c("titration", "sample")], annotation_colors = my_colour)  


sample_cor %>% 
  pheatmap(cluster_rows = TRUE, clustering_distance_cols = "euclidean", 
                clustering_method = "complete", show_colnames = TRUE, show_rownames = T, 
                color = colour_pal, main = "Heatmap - Complete",
                annotation = pDat[c("titration", "sample")], annotation_colors = my_colour)  

```  

##### 3.2.3 Sample-Gene Correaltion with all sequences

```{r, ea heatmap all genes, cache=TRUE}

hm1 <- pheatmap(eDat, cluster_rows = TRUE, clustering_distance_cols = "euclidean", 
                clustering_method = "ward.D2", show_colnames = TRUE, show_rownames = F, 
                color = colour_pal, main = "Heatmap - Ward.D2",
                annotation = pDat[, c("trimester", "sample", "titration")])
```  

##### 3.2.4 Sample-Gene Correaltion with all sequences  

```{r, ea heatmap non zero, cache=TRUE}

hm2 <- pheatmap(eDat2, cluster_rows = TRUE, clustering_distance_cols = "euclidean", 
                clustering_method = "ward.D2", show_colnames = TRUE, show_rownames = F, 
                color = colour_pal, main = "Heatmap - Ward.D2",
                annotation = pDat[, c("trimester", "sample", "titration")])

```  

 ***  

###  4.0 Evaluating by Titration   

I'll now further subset the data according to their titration values.  

####  4.1 Separating eDat by original samples 

```{r, separating by sample}

eDat <- as.data.frame(eDat) %>% 
  rownames_to_column(var ="entrezgene_id") #need column for join later

eDat2 <- as.data.frame(eDat2) %>% 
  rownames_to_column(var ="entrezgene_id")

eDat_s1 <- eDat[c(1:9)] #PL-290v

eDat_s2 <- eDat[c(1, 10:17)] #TCR-2-18_v_SNAP

eDat_con <- eDat[c(1, 18:25)] #GSC control sample FG032A


eDat_s1_del <- eDat2[c(1:9)] #PL-290v

eDat_s2_del <- eDat2[c(1, 10:17)] #TCR-2-18_v_SNAP

eDat_con_del <- eDat2[c(1, 18:25)] #GSC control sample FG032A
```  

####  4.2 Separating eDat by titration levels   

```{r, separating by titration}


#For eDat with all sequences
ng_500 <- list(eDat_con[c(1, 2)], eDat_s2[c(1, 2)], eDat_s1[c(1, 2)])
ng_250 <- list(eDat_con[c(1, 3)], eDat_s2[c(1, 3)], eDat_s1[c(1, 3)])
ng_100 <- list(eDat_con[c(1, 4)], eDat_s2[c(1, 4)], eDat_s1[c(1, 4)])
ng_50 <- list(eDat_con[c(1, 5)], eDat_s2[c(1, 5)], eDat_s1[c(1, 5)])
ng_25 <- list(eDat_con[c(1, 6)], eDat_s2[c(1, 6)], eDat_s1[c(1, 6)])
ng_5 <- list(eDat_con[c(1, 7)], eDat_s2[c(1, 7)], eDat_s1[c(1, 7)])
ng_1 <- list(eDat_con[c(1, 8)], eDat_s2[c(1, 8)], eDat_s1[c(1, 8)])
ng_0 <- list(eDat_con[c(1, 9)], eDat_s2[c(1, 9)], eDat_s1[c(1, 9)])

ng_500 <- join_all(ng_500, by='entrezgene_id', type='left')
ng_250 <- join_all(ng_250, by='entrezgene_id', type='left')
ng_100 <- join_all(ng_100, by='entrezgene_id', type='left')
ng_50 <- join_all(ng_50, by='entrezgene_id', type='left')
ng_25 <- join_all(ng_25, by='entrezgene_id', type='left')
ng_5 <- join_all(ng_5, by='entrezgene_id', type='left')
ng_1 <- join_all(ng_1, by='entrezgene_id', type='left')
ng_0 <- join_all(ng_0, by='entrezgene_id', type='left')


#For eDat with all non-zero seq
ng_500_del <- list(eDat_con_del[c(1, 2)], eDat_s2_del[c(1, 2)], eDat_s1_del[c(1, 2)])
ng_250_del <- list(eDat_con_del[c(1, 3)], eDat_s2_del[c(1, 3)], eDat_s1_del[c(1, 3)])
ng_100_del <- list(eDat_con_del[c(1, 4)], eDat_s2_del[c(1, 4)], eDat_s1_del[c(1, 4)])
ng_50_del <- list(eDat_con_del[c(1, 5)], eDat_s2_del[c(1, 5)], eDat_s1_del[c(1, 5)])
ng_25_del <- list(eDat_con_del[c(1, 6)], eDat_s2_del[c(1, 6)], eDat_s1_del[c(1, 6)])
ng_5_del <- list(eDat_con_del[c(1, 7)], eDat_s2_del[c(1, 7)], eDat_s1_del[c(1, 7)])
ng_1_del <- list(eDat_con_del[c(1, 8)], eDat_s2_del[c(1, 8)], eDat_s1_del[c(1, 8)])
ng_0_del <- list(eDat_con_del[c(1, 9)], eDat_s2_del[c(1, 9)], eDat_s1_del[c(1, 9)])

ng_500_del <- join_all(ng_500_del, by='entrezgene_id', type='left')
ng_250_del <- join_all(ng_250_del, by='entrezgene_id', type='left')
ng_100_del <- join_all(ng_100_del, by='entrezgene_id', type='left')
ng_50_del <- join_all(ng_50_del, by='entrezgene_id', type='left')
ng_25_del <- join_all(ng_25_del, by='entrezgene_id', type='left')
ng_5_del <- join_all(ng_5_del, by='entrezgene_id', type='left')
ng_1_del <- join_all(ng_1_del, by='entrezgene_id', type='left')
ng_0_del <- join_all(ng_0_del, by='entrezgene_id', type='left')

eDat <- as.data.frame(eDat) %>% 
  column_to_rownames(var ="entrezgene_id")

eDat2 <- as.data.frame(eDat2) %>% 
  column_to_rownames(var ="entrezgene_id")
```

####  4.3 Making pDat for titrations   

```{r, making pDats}

pDat_0 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_0))

pDat_1 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_1))

pDat_5 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_5))

pDat_25 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_25))

pDat_50 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_50))

pDat_100 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_100))

pDat_250 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_250))

pDat_500 <- pDat %>% 
  rownames_to_column(var = "GSC_ID") %>% 
  filter(GSC_ID %in% colnames(ng_500))

```  

####  4.4 EA   

Graphing the data now

#####  4.4.1 Non-Filtered eDat Plots 

Frequency plots by titration for all sequences:  

```{r, titration density plots code, echo=FALSE}
 
pl0 <- ng_0 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "0ng Conc", x ="Raw Counts", y = "Frequency") 
  #coord_cartesian(x = c(0, 5000), y = c(0, 30000)) #zooming in

pl1 <- ng_1 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
 ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "1ng Conc ", x ="Raw Counts", y = "Frequency")

pl5 <- ng_5 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  labs(title = "5ng Conc ", x ="Raw Counts", y = "Frequency")
  

pl25 <- ng_25 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "25ng Conc ", x ="Raw Counts", y = "Frequency") 

pl50 <- ng_50 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "50ng Conc ", x ="Raw Counts", y = "Frequency")

pl100 <- ng_100 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "100ng Conc ", x ="Raw Counts", y = "Frequency")

pl250 <- ng_250 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "250ng Conc ", x ="Raw Counts", y = "Frequency") 

```  


```{r, titration denisty plot, fig.height=10, fig.width=18, cache = TRUE}

pl500 <- ng_500 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "500ng Conc ", x ="Raw Counts", y = "Frequency", caption = "Non-Filtered data")
  #coord_cartesian(x = c(0, 5)) #zooming in  

egg::ggarrange(pl0, pl1, pl5, pl25, pl50, pl100, pl250, pl500, nrow = 2)
```  

These plots show that the majority of the sncRNAs have counts equal to zero. To get a better view, I'm going to zoom in the plots having the same x and y axes.  

```{r, titration density plots zoomed, echo=FALSE, cache=TRUE}
 
pl0z <- ng_0 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "0ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl1z <- ng_1 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "1ng Conc ", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl5z <- ng_5 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "5ng Conc ", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl25z <- ng_25 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "25ng Conc ", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl50z <- ng_50 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "50ng Conc ", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl100z <- ng_100 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "bottom") +
  labs(title = "100ng Conc ", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl250z <- ng_250 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "250ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl500z <- ng_500 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "500ng Conc" , subtitle = "Binwidth = 5", 
       x ="Raw Counts", y = "Frequency", caption = "Non-Filtered data; Binwidth = 5") +
  coord_cartesian(x = c(0, 50)) #zooming in  

egg::ggarrange(pl0z, pl1z, pl5z, pl25z, pl50z, pl100z, pl250z, pl500z, nrow = 2)
```  


```{r, 250 and 500 zoomed further, echo = F, eval=FALSE, include=FALSE}

pl250zz <- ng_250 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + #guide turns off legend for this parameter
  theme_minimal() +
  theme(legend.position = "bottom") +
  labs(title = "250ng Conc ", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 100), y = c(0, 3)) #zooming in  

pl500zz <- ng_500 %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "500ng Conc ", x ="Raw Counts", y = "Frequency", caption = "Normalised data") +
  coord_cartesian(x = c(0, 5), y = c(0, 3)) #zooming in  

egg::ggarrange(pl250zz, pl500zz, nrow = 1)

```  

#####  4.4.2 Filtered eDat Plots 

Frequency plots for filtered count df `eDat2` by titration (counts >= 1 in atleast 1 sample): 

```{r, filtered titration density plots zoomed, echo=FALSE}
 

pl0_del <- ng_0_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "0ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl1_del <- ng_1_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "1ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl5_del <- ng_5_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "5ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl25_del <- ng_25_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "25ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl50_del <- ng_50_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "50ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl100_del <- ng_100_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "100ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl250_del <- ng_250_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "250ng Conc", x ="Raw Counts", y = "Frequency") +
  coord_cartesian(x = c(0, 50))

pl500_del <- ng_500_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample, colour = Sample)) +
  geom_histogram(binwidth = 5, alpha = 0.2) +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  scale_colour_manual(values = colpal2, guide = FALSE) + 
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "500ng Conc", x ="Raw Counts", y = "Frequency", caption = "Filtered Data; Binwidth = 5") +
  coord_cartesian(x = c(0, 50))

```  

Again, as above, when plotted by default the data dies not provide good visualisation.   

```{r, zoomed out filtered plot, echo =FALSE, cache=TRUE}

ng_50_del %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Count, fill = Sample)) +
  geom_density(alpha = 0.2, colour = "#606060") +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(title = "50ng Conc ", x ="Raw Counts", y = "Frequency", caption = "Filtered data")
  #coord_cartesian(x = c(0, 5)) #zooming in 

``` 

So, zooming in  

```{r, normalised egg plot, echo=FALSE, cache=TRUE}

egg::ggarrange(pl0_del, pl1_del, pl5_del, pl25_del, pl50_del, pl100_del, pl250_del, pl500_del, nrow = 2)

```  

#####  4.4.3 Gene Specific Expression   {.tabset}  

[This paper](https://www.nature.com/articles/s41598-017-16916-y/tables/1 ) (as with others) mentiones a few housekeeping/ control placental genes for qPCR mRNA experiments. I decided to plot the expression of the sncRNAs annotated to these genes.

But first I had to select which of those showed expression, if any, across all samples to be comparable.  Six genes showed some levels of expression of the 25-30 I searched for:  

+ MRPS18B 
+ IPO8
+ PUM1
+ HSP90AB1
+ LDHA
+ SDHA  

I used the GSC control sample as the one to first test the levels of expression across all of its different tirations.  


```{r, reference genes}

ids_of_int <- c("MRPS18B", "IPO8", "PUM1", "HSP90AB1", "LDHA", "SDHA")
ref_genes_2 <- c("MRPS18B", "IPO8", "PUM1", "LDHA", "SDHA")
ref_genes_3 <- c("MRPS18B", "IPO8", "PUM1", "SDHA")
ref_genes <- c("MRPS18B", "IPO8", "PUM1") 

ids_of_int <- entrez %>% 
  filter(hgnc_symbol %in% ids_of_int)

ref_genes_1 <- ids_of_int$entrezgene_id 

ref_genes_2 <- entrez %>% 
  filter(hgnc_symbol %in% ref_genes_2)
ref_genes_2 <- ref_genes_2$entrezgene_id

ref_genes_3 <- entrez %>% 
  filter(hgnc_symbol %in% ref_genes_3)
ref_genes_3 <- ref_genes_3$entrezgene_id

ref_genes <- entrez %>% 
  filter(hgnc_symbol %in% ref_genes)
ref_genes <- ref_genes$entrezgene_id

sample_names <- pDat$sample
```

###### All genes  

```{r, testing different genes in control sample}

#testing all genes of interest
eDat_con %>% 
  filter(entrezgene_id %in% ref_genes_1) %>% 
  right_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:9, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Sample, y = Count, fill = hgnc_symbol)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  #scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for FG032A", subtitle = "GSC Control Sample", x = "Titration", y = "Count" ) +
  scale_x_discrete(labels=c("m19832" = "0ng", "m19833" = "1ng", "m19834" = "5ng", "m19835" = "25ng", 
                            "m19836" = "50ng", "m19837" = "100ng", "m19838"= "250ng", "m19839" = "500ng")) +
  scale_fill_brewer(palette = "Accent", name = "HGNC Symbol") +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype"))
```

###### 5 Genes

```{r, testing 5 genes of interest, echo=FALSE}

eDat_con %>% 
  filter(entrezgene_id %in% ref_genes_2) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>%  
  pivot_longer(cols = 2:9, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Sample, y = Count, fill = hgnc_symbol)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  #scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for FG032A", subtitle = "GSC Control Sample", x = "Titration", y = "Count" ) +
  scale_x_discrete(labels=c("m19832" = "0ng", "m19833" = "1ng", "m19834" = "5ng", "m19835" = "25ng", 
                            "m19836" = "50ng", "m19837" = "100ng", "m19838"= "250ng", "m19839" = "500ng")) +
  scale_fill_brewer(palette = "Accent", name = "HGNC Symbol") +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype"))
```

###### 4 genes

```{r, testing 4 genes of interest, echo=FALSE}

eDat_con %>% 
  filter(entrezgene_id %in% ref_genes_3) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>%
  pivot_longer(cols = 2:9, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Sample, y = Count, fill = hgnc_symbol)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  #scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for FG032A", subtitle = "GSC Control Sample", x = "Titration", y = "Count" ) +
  scale_x_discrete(labels=c("m19832" = "0ng", "m19833" = "1ng", "m19834" = "5ng", "m19835" = "25ng", 
                            "m19836" = "50ng", "m19837" = "100ng", "m19838"= "250ng", "m19839" = "500ng")) +
  scale_fill_brewer(palette = "Accent", name = "HGNC Symbol") +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype"))

```

###### 3 genes

```{r, testing 3 genes of interest - bar and dot, echo=FALSE}

goi_con <- eDat_con %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>%
  pivot_longer(cols = 2:9, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Sample, y = Count, fill = hgnc_symbol)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  #scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for FG032A - Bar Plot", subtitle = "GSC Control Sample", x = "Titration", y = "Count" ) +
  scale_x_discrete(labels=c("m19832" = "0ng", "m19833" = "1ng", "m19834" = "5ng", "m19835" = "25ng", 
                            "m19836" = "50ng", "m19837" = "100ng", "m19838"= "250ng", "m19839" = "500ng")) +
  scale_fill_brewer(palette = "Accent", name = "HGNC Symbol") +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) 
goi_con

#dot plot
eDat_con %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>%
  pivot_longer(cols = 2:9, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Sample, y = Count, colour = hgnc_symbol)) +
  geom_point(size = 2, position=position_jitter(w=0.1,h=0)) +
  theme_minimal() +
  scale_colour_manual(values = colpal1, name = "HGNC Symbol") +
  labs(title = "Counts for FG032A - Dot Plot", subtitle = "GSC Control Sample", x = "Titration", y = "Count" ) +
  scale_x_discrete(labels=c("m19832" = "0ng", "m19833" = "1ng", "m19834" = "5ng", "m19835" = "25ng", 
                            "m19836" = "50ng", "m19837" = "100ng", "m19838"= "250ng", "m19839" = "500ng")) +
  #scale_fill_brewer(palette = "Accent", name = "HGNC Symbol") +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype"))

```  

#####  Comparing for 3 genes   

These threee genes, *MRPS18B, IPO8", PUM1*, seemed to be comparable. So then I plotted them for the two Rob Lab test samples.

```{r, gene expression by s1 s2, echo=FALSE}

goi_s1 <- eDat_s1 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>%
  pivot_longer(cols = 2:9, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Sample, y = Count, fill = hgnc_symbol)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  #scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for PL-290", subtitle = "Rob Lab T1 Sample", x = "Titration", y = "Count" ) +
  scale_x_discrete(labels=c("m19816" = "0ng", "m19817" = "1ng", "m19818" = "5ng", "m19819" = "25ng", 
                            "m19820" = "50ng", "m19821" = "100ng", "m19822"= "250ng", "m19823" = "500ng")) +
  scale_fill_brewer(palette = "Accent", name = "HGNC Symbol") +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype"),
        legend.position ="none")

goi_s2 <- eDat_s2 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>%
  pivot_longer(cols = 2:9, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = Sample, y = Count, fill = hgnc_symbol)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  #scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for TCR-2-18", subtitle = "Rob Lab Term Sample", x = "Titration", y = "Count" ) +
  scale_x_discrete(labels=c("m19824" = "0ng", "m19825" = "1ng", "m19826" = "5ng", "m19827" = "25ng", 
                            "m19828" = "50ng", "m19829" = "100ng", "m19830"= "250ng", "m19831" = "500ng")) +
  scale_fill_brewer(palette = "Accent", name = "HGNC Symbol") +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype"),
        legend.position ="none")

```  


```{r, compare gene expression, fig.height = 7, fig.width=24}

egg::ggarrange(goi_s1, goi_s2, goi_con, nrow = 1)  

```  


#####  Comparing the 3 genes by titratrion   {.tabset}

Visualising the same data in a different way, this time for each tiration by sample  

###### 0ng

```{r, 0ng gene exp comparison, echo=FALSE}

#Gene expression by same titration

ng_0 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 0ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))

```

###### 1ng

```{r, 1ng gene exp comparison, echo=FALSE}

ng_1 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 1ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))

```

###### 5ng

```{r, 5ng gene exp comparison, echo=FALSE}

ng_5 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 5ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))

```

###### 25ng

```{r, 25ng gene exp comparison, echo=FALSE}

ng_25 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 25ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))

```

###### 50ng

```{r, 50ng gene exp comparison, echo=FALSE}

ng_50 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 50ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))

```

###### 100ng

```{r, 100ng gene exp comparison, echo=FALSE}

ng_100 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 100ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))

```

###### 250ng

```{r, 250ng gene exp comparison, echo=FALSE}

ng_250 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 250ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))

```

###### 500ng

```{r, 500ng gene exp comparison, echo=FALSE}

ng_500 %>% 
  filter(entrezgene_id %in% ref_genes) %>% 
  inner_join(ids_of_int, by = "entrezgene_id") %>% 
  pivot_longer(cols = 2:4, names_to = "Sample", values_to = "Count") %>% 
  ggplot(aes(x = hgnc_symbol, y = Count, fill = Sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts for 500ng Titration", subtitle = "", x = "HGNC Symbol", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype")) +
  coord_cartesian(y = c(0, 50))


```


```{r, plot of rowsums, eval=FALSE, include=FALSE}

eDat_s1 <- eDat [c(1:9)] #PL-290v
eDat_s1 <- eDat_s1 %>% 
  mutate(Sum_s1 = rowSums(eDat_s1[,2:9])) %>% 
  dplyr::arrange(desc(Sum_s1))

eDat_s2 <- eDat [c(1,10:17)] #TCR-2-18_v_SNAP
eDat_s2 <- eDat_s2 %>% 
  mutate(Sum_s2 = rowSums(eDat_s2[,2:9])) %>% 
  dplyr::arrange(desc(Sum_s2))

eDat_con <- eDat [c(1,18:25)] #GSC control sample FG032A
eDat_con <- eDat_con %>% 
  mutate(Sum_con = rowSums(eDat_con[,2:9])) %>% 
  dplyr::arrange(desc(Sum_con))

list <- list(eDat_con[1:10,], eDat_s2, eDat_s1) #make a list of all the dfs to compare. Since I only want to compare against the top 10 hits from the control samples, I've further subsetted eDat_con for only the first 10 rows 

compare <- join_all(list, by='entrezgene_id', type='left') #input the list containg the dfs

compare <- compare %>% 
  dplyr::select("entrezgene_id", "Sum_con", "Sum_s1", "Sum_s2", everything())

list2 <- list(eDat_con[c(1, 10)], eDat_s2[c(1, 10)], eDat_s1[c(1, 10)])
merged <- join_all(list2, by='entrezgene_id', type='left')

merged <- merged %>% 
  filter(Sum_con == Sum_s1) %>% 
  filter(Sum_con == Sum_s2) %>% 
  filter(Sum_con > 0)

merged %>% 
  mutate(Count_Expression = Sum_con) %>% 
  count("Count_Expression") %>% 
  kable() %>% 
  kable_styling(full_width = F)

plot <- compare [c(1:4)]
plot <- plot %>% 
  pivot_longer(cols = 2:4, names_to = "sample")

fonts <- fonttable()

plot %>% 
  ggplot(aes(x = entrezgene_id, y = value, fill = sample)) +
  geom_bar(stat ="identity", position = position_dodge()) +
  theme_minimal() +
  scale_fill_manual(values = colpal2, name = "", labels = c("FG032A", "PL-290v", "TCR-2-18_v_SNAP")) +
  labs(title = "Counts summed over samples", subtitle = "Top 10 hits by max count in Control FG032A samples", x = "Entrez ID", y = "Count" ) +
  theme(plot.title = element_text(family = "Palatino Linotype"),
        plot.subtitle = element_text(family = "Palatino Linotype"),
        legend.text = element_text(family = "Palatino Linotype"),
        axis.title = element_text(family = "Palatino Linotype"))

```  

 ***

####  5.0 Saving Data 

```{r, save Data, eval=FALSE}

saveRDS(fc, file = here::here('data', 'processed', "featurecounts_object_GSC_1790.rds"))
saveRDS(eDat, file = here::here('data', 'processed', "eDat_GSC_1790.rds")) 
saveRDS(pDat, file = here::here('data', 'processed', "pDat_GSC_1790.rds"))
saveRDS(mDat, file = here::here('data', 'processed', "mDat_GSC_1790.rds"))
saveRDS(all_ids, file = here::here('data', 'processed', "sequence_identifiers.rds"))

```  

 ***






